sql: remove some EXPLAIN [ANALYZE] fields in non-VERBOSE mode

mgartner opened the pull request:
The output of `EXPLAIN [ANALYZE]` in non-`VERBOSE` mode is now more succinct.

In the header above the query plan the following changes have been made:

  1. "isolation level" is only shown if it is not "serializable".
  2. "priority" is only shown if it is not "normal".
  3. "quality of service" is only shown if it is not "regular".
  4. "kv cpu time" is not shown.
  5. "sql cpu time" is not shown.
  6. "DistSQL network usage" is only show if it is greater than 0.
  7. "maximum memory usage" is not shown.
  8. "cumulative time spent in KV" is not shown.
  9. "vectorized" is only shown if it is "false".
 10. "buffered writes enabled" is not shown.

In each node of the query plan the following changes have been made:

  1. "KV pairs read" is not shown.
  2. "KV bytes read" is not shown.
  3. "KV gRPC calls" is not shown.
  4. "estimated max memory allocated" is not shown.
  5. "estimated max sql temp disk usage" is not shown.
  6. "actual row count" is directly above "estimated row count".

Here's an example of what `EXPLAIN ANALYZE` looked like before:

```
planning time: 243µs
execution time: 632µs
distribution: local
vectorized: true
plan type: generic, reused
cumulative time spent in KV: 331µs
maximum memory usage: 20 KiB
DistSQL network usage: 0 B (0 messages)
regions: us-east1
sql cpu time: 20µs
estimated RUs consumed: 0
isolation level: serializable
priority: normal
quality of service: regular

• scan
  sql nodes: n1
  kv nodes: n1
  regions: us-east1
  actual row count: 0
  KV time: 331µs
  KV rows decoded: 0
  KV bytes read: 0 B
  KV gRPC calls: 1
  estimated max memory allocated: 20 KiB
  sql cpu time: 20µs
  missing stats
  table: t@t_pkey
  spans: [/0 - /0]
```

And here's the output with the same query after the changes in this commit:

```
planning time: 325µs
execution time: 974µs
distribution: local
plan type: generic, reused
regions: us-east1
estimated RUs consumed: 0

• scan
  sql nodes: n1
  kv nodes: n1
  regions: us-east1
  KV time: 610µs
  KV rows decoded: 0
  sql cpu time: 29µs
  actual row count: 0
  missing stats
  table: t@t_pkey
  spans: [/0 - /0]
```

Epic: None
Release note (sql change): The output of `EXPLAIN [ANALYZE]` in non-`VERBOSE` mode is now more succinct.

mgartner committed:
sql: remove some EXPLAIN [ANALYZE] fields in non-VERBOSE mode

The output of `EXPLAIN [ANALYZE]` in non-`VERBOSE` mode is now more
succinct.

In the header above the query plan the following changes have been made:

  1. "isolation level" is only shown if it is not "serializable".
  2. "priority" is only shown if it is not "normal".
  3. "quality of service" is only shown if it is not "regular".
  4. "kv cpu time" is not shown.
  5. "sql cpu time" is not shown.
  6. "DistSQL network usage" is only show if it is greater than 0.
  7. "maximum memory usage" is not shown.
  8. "cumulative time spent in KV" is not shown.
  9. "vectorized" is only shown if it is "false".
 10. "buffered writes enabled" is not shown.

In each node of the query plan the following changes have been made:

  1. "KV pairs read" is not shown.
  2. "KV bytes read" is not shown.
  3. "KV gRPC calls" is not shown.
  4. "estimated max memory allocated" is not shown.
  5. "estimated max sql temp disk usage" is not shown.
  6. "actual row count" is directly above "estimated row count".

Here's an example of what `EXPLAIN ANALYZE` looked like before:

```
planning time: 243µs
execution time: 632µs
distribution: local
vectorized: true
plan type: generic, reused
cumulative time spent in KV: 331µs
maximum memory usage: 20 KiB
DistSQL network usage: 0 B (0 messages)
regions: us-east1
sql cpu time: 20µs
estimated RUs consumed: 0
isolation level: serializable
priority: normal
quality of service: regular

• scan
  sql nodes: n1
  kv nodes: n1
  regions: us-east1
  actual row count: 0
  KV time: 331µs
  KV rows decoded: 0
  KV bytes read: 0 B
  KV gRPC calls: 1
  estimated max memory allocated: 20 KiB
  sql cpu time: 20µs
  missing stats
  table: t@t_pkey
  spans: [/0 - /0]
```

And here's the output with the same query after the changes in this
commit:

```
planning time: 325µs
execution time: 974µs
distribution: local
plan type: generic, reused
regions: us-east1
estimated RUs consumed: 0

• scan
  sql nodes: n1
  kv nodes: n1
  regions: us-east1
  KV time: 610µs
  KV rows decoded: 0
  sql cpu time: 29µs
  actual row count: 0
  missing stats
  table: t@t_pkey
  spans: [/0 - /0]
```

Release note (sql change): The output of `EXPLAIN [ANALYZE]` in
non-`VERBOSE` mode is now more succinct.

yuzefovich reviewed (approved):
Thanks! I only have one consideration. <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> @yuzefovich reviewed 4 of 4 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/153361#-:-OZx8mlv0-sbf5UVLJWC:bqiqhlk)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @DrewKimball and @kyle-a-wong)

___
*[`-- commits` line 16 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/153361#-OZx8-9X7d2yJAWb2JDh:-OZx8-9Y5QdtiGMJGXMY:blkdufl):*
I wonder whether we should show it if we have `vectorized: false`?


<!-- Sent from Reviewable.io -->

DrewKimball reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thanks for doing this!

> @DrewKimball reviewed 4 of 4 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/153361#-:-OZyzghE91LNIiH5bsVz:b-pt8m11)* status: <span>:shipit:</span> complete! 2 of 0 LGTMs obtained (waiting on @kyle-a-wong and @mgartner)



<!-- Sent from Reviewable.io -->

yuzefovich committed:
logictest: speed up statement_hint_builtins

This test file relies on awaiting for the rangefeed to catch up, so
let's reduce the closed TS interval.

Release note: None

yuzefovich reviewed (approved):
I fixed up a few remaining and new spots.

bors r=yuzefovich,drewkimball

> @yuzefovich reviewed 135 files and all commit messages, made 1 comment, and resolved 1 discussion.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/153361#-:-OiL_yHrCf3Dc8I0xWOE:bu3xcb7)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 2 stale) (waiting on @mgartner).



<!-- Sent from Reviewable.io -->
